polyhost/services/fontpack_render.py

```
@@ -0,0 +1,210 @@
+    contiguous ranges) is drawn as a dim dashed placeholder so it's clearly
+    distinguishable from a black glyph."""
+    g = font.glyphs[cp - font.first]
+    lab_h = 11 if label else 0
+    img = Image.new("L", (cell_w, cell_h + lab_h), 0)
+    draw = ImageDraw.Draw(img)
+    fnt = ImageFont.load_default()
+    is_empty = g["width"] == 0 or g["height"] == 0
+    if is_empty:
+        # dashed grey box + "∅" so an empty entry never looks like a black glyph
+        for x in range(1, cell_w - 1, 4):
+            draw.point((x, 1), fill=70); draw.point((x, cell_h - 2), fill=70)
```

Ruff (0.15.18)

[error] Multiple statements on one line (semicolon)

(E702)